src/components/organisms/UserProfileModal/UserProfileModal.tsx

```diff
@@ -141,25 +139,26 @@ const SuspectedLocation: React.FC<{
   currentVenue: WithId<AnyVenue>;
 }> = ({ user, currentVenue }) => {
   // @debt This will currently load all venues in firebase into memory.. not very efficient
-  useFirestoreConnect("venues");
-  const allVenues = useSelector(orderedVenuesSelector);
+  const { relatedVenues } = useRelatedVenues({
+    currentVenueId: currentVenue.id,
+  });
 
   const suspectedLocation = useMemo(
     () => ({
-      venue: allVenues?.find(
+      venue: relatedVenues?.find(
         (v) =>
           (user.lastSeenIn && user.lastSeenIn[currentVenue?.name ?? ""]) ||
           v.name === user.room
       ),
-      room: allVenues?.find(
+      room: relatedVenues?.find(
         (v) =>
           isVenueWithRooms(v) && v.rooms?.find((r) => r.title === user.room)
       ),
     }),
-    [user, allVenues, currentVenue]
+    [user, relatedVenues, currentVenue]
   );
 
-  if (!user.room || !allVenues) {
+  if (!user.room) {
     return null;
   }
```

Comment:
[NB] I was going to ask if we still need this `return null`, but I think this entire `SuspectedLocation` component is legacy tech debt from the old `Playa` days that will be removed eventually. It's only rendered in this file when `ENABLE_SUSPECTED_LOCATION` is `true`, and it's hardcoded to `false`.

Can you please add an `@debt` comment just before the component definition, something like:

```typescript
/**
 * @debt I believe this is a legacy component leftover from the Playa template/similar outdated code, and should be removed as part of that cleanup
 * @deprecated legacy component, to be removed
 */
const SuspectedLocation: React.FC<{
```

Reply:
Added in 1e8fb879bcd345ad7c00ceb417ac1dfa78c5bbff